lib/base/application.cpp

```diff
@@ -216,41 +236,53 @@ void Application::SetResourceLimits(void)
 		rl.rlim_max = RLIM_INFINITY;
 	}
 
-	if (set_stack_rlimit)
-		rl.rlim_cur = 256 * 1024;
-	else
-		rl.rlim_cur = rl.rlim_max;
+	rlim_t stackLimit;
 
-	if (setrlimit(RLIMIT_STACK, &rl) < 0)
-		Log(LogNotice, "Application", "Could not adjust resource limit for stack size (RLIMIT_STACK)");
-	else if (set_stack_rlimit) {
-		char **new_argv = static_cast<char **>(malloc(sizeof(char *) * (argc + 2)));
+	stackLimit = GetRLimitStack();
 
-		if (!new_argv) {
-			perror("malloc");
-			Exit(EXIT_FAILURE);
+	if (stackLimit != 0) {
+		if (stackLimit < GetDefaultRLimitStack()) {
+			Log(LogWarning, "Application")
+			    << "The user-specified value for RLimitStack cannot be smaller than the default value (" << GetDefaultRLimitStack() << "). Using the default value instead.";
+			stackLimit = GetDefaultRLimitStack();
 		}
 
-		new_argv[0] = argv[0];
-		new_argv[1] = strdup("--no-stack-rlimit");
+		if (set_stack_rlimit)
+			rl.rlim_cur = stackLimit;
+		else
+			rl.rlim_cur = rl.rlim_max;
 
-		if (!new_argv[1]) {
-			perror("strdup");
-			exit(1);
-		}
+		if (setrlimit(RLIMIT_STACK, &rl) < 0)
+			Log(LogNotice, "Application", "Could not adjust resource limit for stack size (RLIMIT_STACK)");
+		else if (set_stack_rlimit) {
```

Comment:
Hello @dnsmichi @lazyfrosch 
Is this condition wrong?
When `setrlimit(RLIMIT_STACK, &rl)` succeeded, why we call `execvp` with `--no-stack-rlimit` next?
I think the `else if (set_stack_rlimit)` should be deleted.

Reply:
Fails with trying to set limits and you don't have permissions to set limits